core/store/service.go

```diff
@@ -0,0 +1,27 @@
+package store
+
+import "context"
+
+// KVStoreService represents a unique, non-forgeable handle to a regular merkle-tree
+// backed KVStore. It should be provided as a module-scoped dependency by the runtime
+// module being used to build the app.
+type KVStoreService interface {
+	// OpenKVStore retrieves the KVStore from the context.
+	OpenKVStore(context.Context) KVStore
```

Comment:
Who is the consumer of the `Open...` APIs?

Reply:
Any module would receive a reference to these services. It's an alternative to using store keys. This could be passed directly to collections for instance instead of a store key and means we don't need to do some weird interface casting on context like we're doing now.

Reply:
Ok so the module's business logic handler, let's call it keeper, would be:

```
type Keeper struct {
    kv core.KVStoreService
    Objects collections.Map[K, V]
}

func (k Keeper) Business(ctx context.Context) {
    store := k.KV.OpenKVStore(ctx)
    k.Objects.Set(store, ...)
}
```
?

Reply:
Yeah, I like that.